File: Forum.Common/Models/PaginateRequest.cs
namespace Forum.Common.Models
{
    public class PaginateRequest
    {
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public string ColumnNameForSorting { get; set; }
        public string SortDirection { get; set; }
        public Filter Filter { get; set; }
    }
}

File: Forum.Angular/src/app/_models/pagination/paginate-request.ts
import { MatPaginator } from "@angular/material/paginator";
import { MatSort } from "@angular/material/sort";
import { Filter } from "./filter";

export class PaginateRequest {
    pageIndex: number;
    pageSize: number;
    columnNameForSorting: string;
    sortDirection: string;
    filter: Filter;

    constructor(matPaginator: MatPaginator, sort: MatSort, filter: Filter) {
        this.pageIndex = matPaginator.pageIndex;
        this.pageSize = matPaginator.pageSize;
        this.columnNameForSorting = sort.active;
        this.sortDirection = sort.direction;
        this.filter = filter;
    }
}